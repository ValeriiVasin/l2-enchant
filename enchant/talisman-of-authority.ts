import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const talismanOfAuthorityEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Drop,
  enchantItem: Item.TalismanOfAuthorityFragment,
  chances: [70, 60, 50, 40, 50, 60, 50, 40, 30, 50, 30, 25, 20, 10, 50, 25, 20, 15, 10],
  items: [
    Item.TalismanOfAuthority_1,
    Item.TalismanOfAuthority_2,
    Item.TalismanOfAuthority_3,
    Item.TalismanOfAuthority_4,
    Item.TalismanOfAuthority_5,
    Item.TalismanOfAuthority_6,
    Item.TalismanOfAuthority_7,
    Item.TalismanOfAuthority_8,
    Item.TalismanOfAuthority_9,
    Item.TalismanOfAuthority_10,
    Item.TalismanOfAuthority_11,
    Item.TalismanOfAuthority_12,
    Item.TalismanOfAuthority_13,
    Item.TalismanOfAuthority_14,
    Item.TalismanOfAuthority_15,
    Item.TalismanOfAuthority_16,
    Item.TalismanOfAuthority_17,
    Item.TalismanOfAuthority_18,
    Item.TalismanOfAuthority_19,
    Item.TalismanOfAuthority_20,
  ],
  boundaries: new Set([Item.TalismanOfAuthority_6, Item.TalismanOfAuthority_11, Item.TalismanOfAuthority_16]),
});
