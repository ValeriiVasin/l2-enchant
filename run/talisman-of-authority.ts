import { header } from '../helpers/header';
import { run } from '../helpers/run';
import { Item } from '../items';

const times = 10_000;

console.log();
console.log(header('Талисман Властителя'));
run(
  Item.TalismanOfAuthority_1,
  [
    Item.TalismanOfAuthority_2,
    Item.TalismanOfAuthority_3,
    Item.TalismanOfAuthority_4,
    Item.TalismanOfAuthority_5,
    Item.NobleTalismanOfAuthority_1,
  ],
  times,
);

console.log();
console.log(header('Талисман Благородного Властителя'));
run(
  Item.NobleTalismanOfAuthority_1,
  [
    Item.NobleTalismanOfAuthority_2,
    Item.NobleTalismanOfAuthority_3,
    Item.NobleTalismanOfAuthority_4,
    Item.NobleTalismanOfAuthority_5,
    Item.RadiantTalismanOfAuthority_1,
  ],
  times,
);

console.log();
console.log(header('Искрящийся Талисман Властителя'));
run(
  Item.RadiantTalismanOfAuthority_1,
  [
    Item.RadiantTalismanOfAuthority_2,
    Item.RadiantTalismanOfAuthority_3,
    Item.RadiantTalismanOfAuthority_4,
    Item.RadiantTalismanOfAuthority_5,
  ],
  times,
);
