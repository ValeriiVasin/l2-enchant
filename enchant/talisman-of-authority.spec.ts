import { talismanOfAuthorityEnchantMap } from './talisman-of-authority';

it('enchant map: talisman of authority', () => {
  expect(talismanOfAuthorityEnchantMap).toMatchSnapshot();
});
