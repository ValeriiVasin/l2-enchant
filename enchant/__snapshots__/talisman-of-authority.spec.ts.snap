// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: talisman of authority 1`] = `
Map {
  "Талисман Властителя Ур.1" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.1",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Талисман Властителя Ур.2",
      },
      Object {
        "chance": 30,
        "item": "Талисман Властителя Ур.1",
      },
    ],
  },
  "Талисман Властителя Ур.2" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.2",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Талисман Властителя Ур.3",
      },
      Object {
        "chance": 40,
        "item": "Талисман Властителя Ур.1",
      },
    ],
  },
  "Талисман Властителя Ур.3" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.3",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.4",
      },
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.1",
      },
    ],
  },
  "Талисман Властителя Ур.4" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.4",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Властителя Ур.5",
      },
      Object {
        "chance": 60,
        "item": "Талисман Властителя Ур.1",
      },
    ],
  },
  "Талисман Властителя Ур.5" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.5",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.6",
      },
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.1",
      },
    ],
  },
  "Талисман Властителя Ур.6" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.6",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Талисман Властителя Ур.7",
      },
      Object {
        "chance": 40,
        "item": "Талисман Властителя Ур.6",
      },
    ],
  },
  "Талисман Властителя Ур.7" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.7",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.8",
      },
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.6",
      },
    ],
  },
  "Талисман Властителя Ур.8" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.8",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Властителя Ур.9",
      },
      Object {
        "chance": 60,
        "item": "Талисман Властителя Ур.6",
      },
    ],
  },
  "Талисман Властителя Ур.9" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.9",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Властителя Ур.10",
      },
      Object {
        "chance": 70,
        "item": "Талисман Властителя Ур.6",
      },
    ],
  },
  "Талисман Властителя Ур.10" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.10",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.11",
      },
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.6",
      },
    ],
  },
  "Талисман Властителя Ур.11" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.11",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Властителя Ур.12",
      },
      Object {
        "chance": 70,
        "item": "Талисман Властителя Ур.11",
      },
    ],
  },
  "Талисман Властителя Ур.12" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.12",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Талисман Властителя Ур.13",
      },
      Object {
        "chance": 75,
        "item": "Талисман Властителя Ур.11",
      },
    ],
  },
  "Талисман Властителя Ур.13" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.13",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Талисман Властителя Ур.14",
      },
      Object {
        "chance": 80,
        "item": "Талисман Властителя Ур.11",
      },
    ],
  },
  "Талисман Властителя Ур.14" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.14",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Талисман Властителя Ур.15",
      },
      Object {
        "chance": 90,
        "item": "Талисман Властителя Ур.11",
      },
    ],
  },
  "Талисман Властителя Ур.15" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.15",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.16",
      },
      Object {
        "chance": 50,
        "item": "Талисман Властителя Ур.11",
      },
    ],
  },
  "Талисман Властителя Ур.16" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.16",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Талисман Властителя Ур.17",
      },
      Object {
        "chance": 75,
        "item": "Талисман Властителя Ур.16",
      },
    ],
  },
  "Талисман Властителя Ур.17" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.17",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Талисман Властителя Ур.18",
      },
      Object {
        "chance": 80,
        "item": "Талисман Властителя Ур.16",
      },
    ],
  },
  "Талисман Властителя Ур.18" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.18",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Талисман Властителя Ур.19",
      },
      Object {
        "chance": 85,
        "item": "Талисман Властителя Ур.16",
      },
    ],
  },
  "Талисман Властителя Ур.19" => Object {
    "cost": "Осколок Талисмана Властителя",
    "item": "Талисман Властителя Ур.19",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Талисман Властителя Ур.20",
      },
      Object {
        "chance": 90,
        "item": "Талисман Властителя Ур.16",
      },
    ],
  },
}
`;
